clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -193,6 +193,30 @@ class CIRCastOpLowering : public mlir::OpConversionPattern<mlir::cir::CastOp> {
           mlir::cir::CmpOpKind::ne, src, zero);
       break;
     }
+    case mlir::cir::CastKind::integral: {
+      auto oldSourceType =
+          castOp->getOperands().front().getType().cast<mlir::IntegerType>();
+      auto sourceValue = adaptor.getOperands().front();
+      auto sourceType = sourceValue.getType().cast<mlir::IntegerType>();
+      auto targetType = getTypeConverter()
+                            ->convertType(castOp.getResult().getType())
+                            .cast<mlir::IntegerType>();
+
+      // target integer is smaller: truncate source value.
+      if (targetType.getWidth() < sourceType.getWidth()) {
+        rewriter.replaceOpWithNewOp<mlir::LLVM::TruncOp>(castOp, targetType,
+                                                         sourceValue);
+      } else {
+        // FIXME: CIR codegen does not distiguishes singned/unsinged types.
```

Comment:
This would be a good thing to fix next (after this PR), can you create an issue to track it?